napari/_qt/dialogs/_tests/test_qt_plugin_dialog.py

```diff
@@ -149,3 +235,90 @@ def test_constructor_visible_widgets(plugin_dialog_constructor):
     """
     assert not plugin_dialog_constructor.direct_entry_edit.isVisible()
     assert not plugin_dialog_constructor.direct_entry_btn.isVisible()
+
+
+def test_version_dropdown(plugin_dialog):
+    """
+    Test that when the source drop down is changed, it displays the other versions properly.
+    """
+    plugin_dialog.available_list.item(
+        1
+    ).widget.source_choice_dropdown.setCurrentIndex(1)
+
+    assert (
+        plugin_dialog.available_list.item(
+            1
+        ).widget.version_choice_dropdown.currentText()
+        == '4.5'
+    )
+
+
+def test_plugin_list_item(plugin_dialog):
+    assert plugin_dialog.installed_list._count_visible() == 2
+
+
+def test_plugin_list_handle_action(plugin_dialog, qtbot):
+    item = plugin_dialog.installed_list.item(0)
+
+    plugin_dialog.installed_list.handle_action(
+        item,
+        'test-name-1',
+        InstallerActions.UPGRADE,
+    )
+
+    with patch.object(qt_plugin_dialog.WarnPopup, "exec_") as mock:
+        plugin_dialog.installed_list.handle_action(
+            item,
+            'test-name-1',
+            InstallerActions.UNINSTALL,
+        )
+        assert mock.called
+
+    item = plugin_dialog.available_list.item(0)
+    plugin_dialog.available_list.handle_action(
+        item,
+        'test-name-1',
+        InstallerActions.INSTALL,
+        version='3',
+    )
+
+    plugin_dialog.available_list.handle_action(
+        item, 'test-name-1', InstallerActions.CANCEL, version='3'
+    )
+
+    # Wait for refresh timer, state and worker to be done
+    qtbot.waitUntil(
+        lambda: not plugin_dialog._add_items_timer.isActive()
+        and plugin_dialog.refresh_state == qt_plugin_dialog.RefreshState.DONE
+    )
+    qtbot.waitUntil(lambda: not plugin_dialog.worker.is_running)
+
+
+def test_on_enabled_checkbox(plugin_dialog, qtbot):
+    # checks npe2 lines
+    item = plugin_dialog.installed_list.item(0)
+    widget = plugin_dialog.installed_list.itemWidget(item)
+
+    with qtbot.waitSignal(widget.enabled_checkbox.stateChanged, timeout=500):
+        widget.enabled_checkbox.setChecked(False)
+
+    # checks npe1 lines
+    item = plugin_dialog.installed_list.item(1)
+    widget = plugin_dialog.installed_list.itemWidget(item)
+
+    with qtbot.waitSignal(widget.enabled_checkbox.stateChanged, timeout=500):
+        widget.enabled_checkbox.setChecked(False)
+
+
+def test_add_items_outdate(plugin_dialog):
+    new_plugin = (
+        npe2.PackageMetadata(name="my-plugin", version="0.3.0"),
+        True,
+        {
+            "home_page": 'www.mywebsite.com',
+            "pypi_versions": ['0.3.0'],
+            "conda_versions": ['0.3.0'],
+        },
+    )
+    plugin_dialog._plugin_data = [new_plugin]
+    plugin_dialog._add_items()
```

Comment:
I do not see any assertions here. What is the expected change produced by this test?

Reply:
I am adding an assertion here.  This is making sure that a plugin is not added to the available plugins list if its already in the installed list.